sdk/index.ts

```diff
@@ -0,0 +1,273 @@
+        defaultFlagHandler?: (featureName: string) => DefaultFlag;
+    }) {
+        this.environmentKey = data.environmentKey;
+        this.apiUrl = data.apiUrl || this.apiUrl;
+        this.customHeaders = data.customHeaders;
+        this.requestTimeoutSeconds = data.requestTimeoutSeconds;
+        this.enableLocalEvaluation = data.enableLocalEvaluation;
+        this.environmentRefreshIntervalSeconds =
+            data.environmentRefreshIntervalSeconds || this.environmentRefreshIntervalSeconds;
+        this.retries = data.retries;
+        this.enableAnalytics = data.enableAnalytics || false;
+        this.defaultFlagHandler = data.defaultFlagHandler;
+
+        this.environmentFlagsUrl = `${this.apiUrl}flags/`;
+        this.identitiesUrl = `${this.apiUrl}identities/`;
+        this.environmentUrl = `${this.apiUrl}environment-document/`;
+
+        this.environment;
+        if (this.enableLocalEvaluation) {
+            this.environmentDataPollingManager = new EnvironmentDataPollingManager(
+                this,
+                this.environmentRefreshIntervalSeconds
+            );
+            this.environmentDataPollingManager.start();
+        }
+
+        this.analyticsProcessor = data.enableAnalytics
+            ? new AnalyticsProcessor({
+                  environmentKey: this.environmentKey,
+                  baseApiUrl: this.apiUrl,
+                  timeout: this.requestTimeoutSeconds
+              })
+            : undefined;
+    }
+    /**
+     * Get all the default for flags for the current environment.
+     *
+     * @returns Flags object holding all the flags for the current environment.
+     */
+    async getEnvironmentFlags(): Promise<Flags> {
+        if (this.environment) {
+            return new Promise(resolve => resolve(this.getEnvironmentFlagsFromDocument()));
+        }
+
+        return this.getEnvironmentFlagsFromApi();
+    }
+    /**
+     * Get all the flags for the current environment for a given identity. Will also
+        upsert all traits to the Flagsmith API for future evaluations. Providing a
+        trait with a value of None will remove the trait from the identity if it exists.
+     * 
+     * @param  {string} identifier a unique identifier for the identity in the current
+            environment, e.g. email address, username, uuid
+     * @param  {{[key:string]:any}} traits? a dictionary of traits to add / update on the identity in
+            Flagsmith, e.g. {"num_orders": 10}
+     * @returns Flags object holding all the flags for the given identity.
+     */
+    getIdentityFlags(identifier: string, traits?: { [key: string]: any }): Promise<Flags> {
+        traits = traits || {};
+        if (this.environment) {
+            return new Promise(resolve =>
+                resolve(this.getIdentityFlagsFromDocument(identifier, traits || {}))
+            );
+        }
+        return this.getIdentityFlagsFromApi(identifier, traits);
+    }
+
+    async update_environment() {
```

Comment:
Minor: This should be camelcase. Since this isn't private I think adding some comments here would be useful in informing people that this could be used to bypass the cache for refreshIntervalSeconds.